composer/profiler/system_profiler.py

```diff
@@ -32,73 +32,75 @@ def __init__(self,
                  profile_disk: bool = False,
                  profile_net: bool = False,
                  stats_thread_interval_seconds: float = 0.5) -> None:
-        self.hparams = SystemProfilerHparams(
-            profile_cpu=profile_cpu,
-            profile_disk=profile_disk,
-            profile_memory=profile_memory,
-            profile_net=profile_net,
-            stats_thread_interval_seconds=stats_thread_interval_seconds,
-        )
+
+        self.profile_cpu = profile_cpu
+        self.profile_disk = profile_disk
+        self.profile_memory = profile_memory
+        self.profile_net = profile_net
+        self.stats_thread_interval_seconds = stats_thread_interval_seconds
 
         try:
             # Attempt an import of psutil in init to ensure it is installed
             import psutil
             del psutil
         except ImportError as e:
-            raise ImportError("Please install composer with pip install composer[perf] to use the profiler") from e
+            raise ImportError(
+                "Please install composer with pip install composer[perf] to use the state.profiler") from e
 
     def init(self, state: State, logger: Logger):
         del logger  # unused
         if state.profiler is None:
             raise RuntimeError(
                 textwrap.dedent("""\
-                    To use the dataloader profiler, state.profiler must be set.
-                    Make sure to run composer with the profiler -- i.e. with the `--profiler` CLI flag."""))
+                    To use the dataloader state.profiler, state.profiler must be set.
+                    Make sure to run composer with the state.profiler -- i.e. with the `--profiler` CLI flag."""))
 
         # Start the stats thread
-        threading.Thread(target=self._stats_thread, daemon=True, args=[state.profiler]).start()
+        threading.Thread(target=self._stats_thread, daemon=True, args=[state]).start()
 
-    def _stats_thread(self, profiler: Profiler):
+    def _stats_thread(self, state: State):
```

Comment:
I think just the profiler needs to be passed through? (No need for the entire state since the markers no longer require it?)